internal/cmd/skupper/link/kube/link_export.go

```diff
@@ -0,0 +1,286 @@
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("link name is not valid: %s", err))
+		}
+
+		cmd.outputFile = args[1]
+	}
+
+	if cmd.flags.tlsSecret == "" {
+		validationErrors = append(validationErrors, fmt.Errorf("the TLS secret name was not specified"))
+	} else {
+		ok, err := resourceStringValidator.Evaluate(cmd.flags.tlsSecret)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("the name of the tls secret is not valid: %s", err))
+		}
+	}
+
+	selectedCost, err := strconv.Atoi(cmd.flags.cost)
+	if err != nil {
+		validationErrors = append(validationErrors, fmt.Errorf("link cost is not valid: %s", err))
+	}
+	ok, err := numberValidator.Evaluate(selectedCost)
+	if !ok {
+		validationErrors = append(validationErrors, fmt.Errorf("link cost is not valid: %s", err))
+	}
+
+	if cmd.flags.output != "" {
+		ok, err := outputTypeValidator.Evaluate(cmd.flags.output)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("output type is not valid: %s", err))
+		}
+	}
+
+	return validationErrors
+}
+
+func (cmd *CmdLinkExport) InputToOptions() {
+
+	cmd.cost, _ = strconv.Atoi(cmd.flags.cost)
+	cmd.tlsSecret = cmd.flags.tlsSecret
+	cmd.output = cmd.flags.output
+	cmd.generateCredential = cmd.flags.generateCredential
+
+}
+
+func (cmd *CmdLinkExport) Run() error {
+
+	if cmd.activeSite == nil {
+		return fmt.Errorf("there is no active site to generate the link resource file")
+	} else if len(cmd.activeSite.Status.Endpoints) == 0 {
+		return fmt.Errorf("the active site has not configured endpoints yet")
+	}
+
+	if cmd.output == "" {
+		return fmt.Errorf("output format has not been specified")
+	}
+
+	resource := v1alpha1.Link{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "skupper.io/v1alpha1",
+			Kind:       "Link",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: cmd.linkName,
+		},
+		Spec: v1alpha1.LinkSpec{
+			TlsCredentials: cmd.tlsSecret,
+			Cost:           cmd.cost,
+			Endpoints:      cmd.activeSite.Status.Endpoints,
+		},
+	}
+
+	cmd.generatedLink = resource
+
+	if cmd.generateCredential {
+		certificate := v1alpha1.Certificate{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "skupper.io/v1alpha1",
+				Kind:       "Certificate",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: cmd.tlsSecret,
+			},
+			Spec: v1alpha1.CertificateSpec{
+				Ca:      cmd.activeSite.Status.DefaultIssuer,
+				Client:  true,
+				Subject: GetSubjectsFromEndpoints(cmd.activeSite.Status.Endpoints),
+			},
+		}
+
+		_, err := cmd.Client.Certificates(cmd.Namespace).Create(context.TODO(), &certificate, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+
+	}
+
+	return nil
+}
+
+func (cmd *CmdLinkExport) WaitUntilReady() error {
+
+	var resourcesToPrint []string
+	encodedOutput, err := utils.Encode(cmd.output, cmd.generatedLink)
+	if err != nil {
+		return err
+	}
+
+	if cmd.generateCredential {
+
+		err := utils.NewSpinner("Waiting for secret to be generated...", 20, func() error {
```

Comment:
@lynnemorrison Message here is different because it is generating a yaml file based on a secret